streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -517,7 +514,9 @@ private KafkaStreams(final InternalTopologyBuilder internalTopologyBuilder,
         if (clientId.length() <= 0)
             clientId = applicationId + "-" + processId;
 
-        this.logPrefix = String.format("stream-client [%s]", clientId);
+        LogContext logContext = new LogContext("[stream-client=" + clientId + "] ");
```

Comment:
Just a general comment: we recommend using `final` for any local variable that would not change.

Reply:
Also: why we want to change the prefix to `stream-client XX` to `stream-client=XX`? A potential risk is that if users depend on the output string prefix constant when greping / debugging the logs it would break their script.

Reply:
Thanks for the above comments @guozhangwang , fixed both. Agreed with risk involved when changing prefix.

Reply:
There's a benefit in using consistent logging across Kafka clients and APIs. The way Jason and I were thinking of this is what you can add key/value keys to the LogContext and they get printed in a consistent way. We don't offer compatibility of the log format, so I don't see that as being a problem, personally.

Reply:
I was not aware of this key-value prefix in LogContext; is it going to be added as a future work post 1.0.0 or before it?

Reply:
And yes we do not make any commitment about log format, still in the past there have been some complaints from users when we changed the log4j format causing their debugging script tooling to break. If we are going to have this key-value typed LogContext in 1.0.0 and we are going to add it as part of the upgrade guide web docs, I'm happy to change to get consistency benefits across clients, but just clarifying the risk.